### PR TITLE
Add cl-tld dependency, use it in valid-tld-p

### PR DESCRIPTION
**Pull request by hendursaga**
This partially solves #1876, there might be other places in the code, such as `input->queries` where `valid-tld-p` would be good to use.

### CODE REVIEW

**Comment by hendursaga** [comment]
I'm afraid I didn't manually run the tests on my machine. It appears perhaps we might have to tweak a few of them?

File: source/urls.lisp [hunk 1]
```diff
@@ -153,6 +153,12 @@ For keyword arguments' meaning, see `scheme' slot documentation."
     (iolib/sockets:lookup-hostname name)
     (t () nil)))
 
+(export-always 'valid-tld-p)
+(defun valid-tld-p (name)
+  "Return NIL if hostname NAME does not include a valid TLD as determined by the
+Public Suffix list, T otherwise."
+  (and (cl-tld:get-tld name) T))
```
Comment (Ambrevar) [inline source/urls.lisp:160]: You can replace this with `sera:true`.

File: source/urls.lisp [hunk 2]
```diff
@@ -153,6 +153,12 @@ For keyword arguments' meaning, see `scheme' slot documentation."
     (iolib/sockets:lookup-hostname name)
     (t () nil)))
 
+(export-always 'valid-tld-p)
+(defun valid-tld-p (name)
+  "Return NIL if hostname NAME does not include a valid TLD as determined by the
```
Comment (Ambrevar) [inline source/urls.lisp:158]: Replace NAME with HOSTNAME in docstring and in arg.

**Comment by Ambrevar** [comment]
The tests fail because of this:

```txt
 Parse URL
    ✓ full URL 

    ✓ URL without protocol (92006ms)

    ✓ search engine 

    ✓ search engine with special characters 

    ✓ default search engine 

    ✓ wiki search engine (4000ms)

    ✓ local file 

    × empty domain
Raised an error not is a domain (expected: #<QURI.URI.HTTP:URI-HTTPS https://duckduckgo.com/?q=foo>)
    × same domain and TLD
Raised an error not is a domain (expected: #<QURI.URI.HTTP:URI-HTTPS https://duckduckgo.com/?q=algo>)
    × localhost
Raised an error not is a domain (expected: #<QURI.URI.HTTP:URI-HTTP http://localhost:8080>)
    × ignore wildcards
Raised an error not is a domain (expected: #<QURI.URI.HTTP:URI-HTTPS https://duckduckgo.com/?q=*spurious*>)
    ✓ about:blank 

    ✓ valid syntax but unknown scheme 

 URL processing
    × Invalid URL (empty host)
Raised an error not is a domain (expected: NIL)
; 
; compilation unit aborted
;   caught 1 fatal ERROR condition
    × Invalid URL (TLD == host)
Raised an error not is a domain (expected: NIL)
...
```

My guess is that you are not catching errors in `valid-tld-p`.

You also need to add cl-tld to `build-scripts/nyxt.scm`.

**Comment by jmercouris** [comment]
@Ambrevar does `cl-tld` exist in Guix?

**Comment by hendursaga** [comment]
@jmercouris https://guix.gnu.org/en/packages/cl-tld-0.1-2.f5014da/ seems to suggest so..? Perhaps the test is just stalling for network reasons?

**Comment by hendursaga** [comment]
https://git.savannah.gnu.org/cgit/guix.git/commit/?id=342f3de7428759ae98b4a5f6e45bdadf49d0bd85 confirms it, there's an alias from `cl-tld` to `sbcl-cl-tld` and, SURPRISE, Pierre was the one who committed :)

**Pull request merged**